### DISCUSSION TITLE
Not able to auth after accessing the IDC app

### DISCUSSION THREAD

**Discussion by surbhigoel948** [aws-samples/iam-identity-center-team#206] [Q&A]
I am not able to access the TEAM console but the page is then loading and stops at the home page instead of giving me access to the tool.
![Screenshot 2024-03-19 at 4 55 28 PM](https://github.com/aws-samples/iam-identity-center-team/assets/66158797/f2ae1419-a55d-4ba2-8283-d5cfe20b428a)
I have done the setup as documented. There were no errors. I am not using the custom domain.

I checked the network trace,
the failure is on request, POST https://XXX-main.auth.us-east-1.amazoncognito.com/oauth2/token
It returns 400 error code
Payload is as follows:
grant_type: authorization_code
code: XXXX
client_id: XXXX
redirect_uri: https://portal.teamtest.online/
Response:
{"error":"unauthorized_client"}
Client ID is the WebClient ID. 
If I click at the federated sign it, it says https://XXXX-main.auth.us-east-1.amazoncognito.com/error?error=redirect_mismatch&client_id=XXXX

Cloudtrail log:
```
{
    "eventVersion": "1.09",
    "userIdentity": {
        "accountId": "XXXX"
    },
    "eventTime": "2024-03-19T23:02:12Z",
    "eventSource": "cognito-idp.amazonaws.com",
    "eventName": "Token_POST",
    "awsRegion": "us-east-1",
    "sourceIPAddress": "XXXXX",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36",
    "errorCode": "400",
    "requestParameters": null,
    "responseElements": null,
    "additionalEventData": {
        "responseParameters": {
            "status": 400
        },
        "requestParameters": {
            "code": [
                "HIDDEN_DUE_TO_SECURITY_REASONS"
            ],
            "grant_type": [
                "authorization_code"
            ],
            "redirect_uri": [
                "https://portal.teamtest.online/"
            ],
            "client_id": [
                "XXXXXXX"
            ]
        },
        "userPoolDomain": "XXXXX-main.auth.us-east-1.amazoncognito.com",
        "userPoolId": "us-east-1_6gv2tw6Na"
    },
    "requestID": "4cb58b33-dcf8-4d1b-847e-7abf283caf31",
    "eventID": "e30879ed-88de-4ad9-b35b-084d4eae0306",
    "readOnly": false,
    "eventType": "AwsServiceEvent",
    "managementEvent": true,
    "recipientAccountId": "XXXX",
    "serviceEventDetails": {
        "serviceAccountId": "745623467555"
    },
    "eventCategory": "Management",
    "tlsDetails": {
        "tlsVersion": "TLSv1.3",
        "cipherSuite": "TLS_AES_128_GCM_SHA256",
        "clientProvidedHostHeader": "XXXXX-main.auth.us-east-1.amazoncognito.com"
    }
}
```

I also checked other discussion, my Identity center and TEAM delegated accounts are same and separate from all other accounts.
Region is same for both, us-east-1

Any help will be appreciated, thanks!

**Comment by hty890614** [comment]
is any update of this issue?
I have the same issue

**Reply by hty890614** [reply]
I have soluved it.
IdentityCenter and cognito should be in the same region
